service/create/load_balancer.go

```diff
@@ -14,28 +14,36 @@ import (
 )
 
 type LoadBalancerInput struct {
-	Clients         awsutil.Clients
-	Cluster         awstpr.CustomObject
-	InstanceIDs     []string
+	// Name must be unique.
+	Name string
+	// TODO can this be a single client?
+	Clients awsutil.Clients
+	Cluster awstpr.CustomObject
+	// InstanceIDs are the IDs of the instances that should be registered with the ELB.
+	InstanceIDs []string
+	// PortsToOpen are the ports the ELB should listen to and forward on.
+	PortsToOpen     []int
 	SecurityGroupID string
 	SubnetID        string
 }
 
-func (s *Service) createLoadBalancer(input LoadBalancerInput) error {
+func (s *Service) createLoadBalancer(input LoadBalancerInput) (*awsresources.ELB, error) {
+	lbName, err := loadBalancerName(input.Name, input.Cluster)
+	if err != nil {
+		return nil, microerror.MaskAny(err)
+	}
+
 	lb := &awsresources.ELB{
-		Name:          input.Cluster.Spec.Cluster.Cluster.ID,
+		Name:          lbName,
 		SecurityGroup: input.SecurityGroupID,
 		SubnetID:      input.SubnetID,
-		PortsToOpen: []int{
-			input.Cluster.Spec.Cluster.Kubernetes.API.SecurePort,
-			input.Cluster.Spec.Cluster.Etcd.Port,
-		},
-		Client: input.Clients.ELB,
+		PortsToOpen:   input.PortsToOpen,
+		Client:        input.Clients.ELB,
 	}
 
 	lbCreated, err := lb.CreateIfNotExists()
 	if err != nil {
-		return microerror.MaskAny(fmt.Errorf("could not create ELB: %s", errgo.Details(err)))
+		return nil, microerror.MaskAny(fmt.Errorf("could not create ELB: %s", errgo.Details(err)))
```

Comment:
Complicated error handling. That also repeats below. I will not repeat these comments. Lets make sure we clean this up.